Support for HA Alarm entity & arm states

vitalijssilins opened the pull request:
This pull request allows you to integrate your home alarm system through Alarmo integration. It allows to specify Alarmo entity panel and arm states that act as filters as to when to send notifications. The code respects the presence filter and works together. You can specify the presence filter and/or Alarmo entity.

This helped me to unblock several additional use cases:

1.  When I & my family are away and someone is looking after the home, it does not send us a notification every time the person is in camera fields, only if the home is armed. (previously, I had to disable automation)
2. If the home is armed for a night, I receive notification if someone is on the territory (duplicate of the automation with presence filter off, but respecting "night arm")
3. When both the presence filter and Alarmo are enabled, it doesn't send any notifications to you while you are leaving or entering the house even if an alarm is set.

AngellusMortis commented:
The updates should not make any mention to Alarmo or any other specific alarm brand/integration. It should solely just support [alarm entities](https://developers.home-assistant.io/docs/core/entity/alarm-control-panel/) .

krohrsb commented:
It should also be optional if possible.

vitalijssilins commented:
@AngellusMortis & @krohrsb improved the code based on your comments.

- Changed Alarmo branded entity selector to HA generic one
- Added "(Optional)" title to alarm entities – it was optional in logic before

AngellusMortis commented on blueprints/automation/unifiprotect/notification_smart_motion_event.yaml:
```diff
@@ -52,6 +57,26 @@ blueprint:
+          domain: alarm_control_panel
+
+    alarm_arm_states:
+      name: (Optional) Alarm Arm States
```
Should be required with a sensible default. Otherwise, if the alarm entity is set and the arm states is not, it will never trigger.

vitalijssilins replied:
Added defaults, changed the condition logic and improved docs.

AngellusMortis commented:
I merged this because I kind of felt bad for letting it sit for so long. But are you able to apply the same changes to the other two blueprints?

vitalijssilins commented:
> I merged this because I kind of felt bad for letting it sit for so long. But are you able to apply the same changes to the other two blueprints?

No worries, I've used it directly, but now I can reapply the official version. Thank you for the merge.
I was not able to apply it to the other two blueprints, as I don't have according hardware to test.
